lib/AST/ASTPrinter.cpp

```diff
@@ -3492,16 +3474,16 @@ void PrintAST::visitFuncDecl(FuncDecl *decl) {
                                                SourceRange(StartLoc, EndLoc));
     printSourceRange(Range, Ctx);
   } else {
-    if (!Options.SkipIntroducerKeywords) {
-      if (decl->isStatic() && Options.PrintStaticKeyword)
-        printStaticKeyword(decl->getCorrectStaticSpelling());
+    if (decl->isStatic() && Options.PrintStaticKeyword)
+      printStaticKeyword(decl->getCorrectStaticSpelling());
 
-      printMutabilityModifiersIfNeeded(decl);
-      if (decl->isConsuming() && !decl->getAttrs().hasAttribute<ConsumingAttr>()) {
-        Printer.printKeyword("__consuming", Options, " ");
-      }
-      Printer << tok::kw_func << " ";
+    printMutabilityModifiersIfNeeded(decl);
```

Comment:
Will these now get printed in places they didn't used to due to not checking `SkipIntroducerKeywords`?  Not sure if that's desirable or not 🤔

Reply:
Renamed `printMutabilityModifiersIfNeeded()` to `printSelfAccessKindModifiersIfNeeded()` and moved `__consuming` printing into it. Also, added a printing option whether to print `mutating`/`nonmutating`/`__consuming`.

So far `SkipIntroducerKeywords` is used only in `<RelatedName />` printing in SourceKit.